tests/mesh/mesh_smoother_test.C

```diff
@@ -81,52 +127,98 @@ public:
     DistortSquare ds;
     MeshTools::Modification::redistribute(mesh, ds);
 
-    // Assert the distortion is as expected
-    auto center_distortion_is = []
+    const auto & boundary_info = mesh.get_boundary_info();
+
+    // Function to assert the distortion is as expected
+    auto center_distortion_is = [&boundary_info]
       (const Node & node, int d, bool distortion,
        Real distortion_tol=TOLERANCE) {
       const Real r = node(d);
       const Real R = r * n_elems_per_side;
-      CPPUNIT_ASSERT_GREATER(-TOLERANCE*TOLERANCE, r);
-      CPPUNIT_ASSERT_GREATER(-TOLERANCE*TOLERANCE, 1-r);
-
-      // If we're at the boundaries we should *never* be distorted
-      if (std::abs(node(0)) < TOLERANCE*TOLERANCE ||
-          std::abs(node(0)-1) < TOLERANCE*TOLERANCE)
-        {
-          const Real R1 = node(1) * n_elems_per_side;
-          CPPUNIT_ASSERT_LESS(TOLERANCE*TOLERANCE, std::abs(R1-std::round(R1)));
-          return true;
-        }
-
-      if (std::abs(node(1)) < TOLERANCE*TOLERANCE ||
-          std::abs(node(1)-1) < TOLERANCE*TOLERANCE)
-        {
-          const Real R0 = node(0) * n_elems_per_side;
-          CPPUNIT_ASSERT_LESS(TOLERANCE*TOLERANCE, std::abs(R0-std::round(R0)));
-
-          return true;
-        }
+      CPPUNIT_ASSERT_GREATER(-distortion_tol*distortion_tol, r);
+      CPPUNIT_ASSERT_GREATER(-distortion_tol*distortion_tol, 1-r);
 
       // If we're at the center we're fine
-      if (std::abs(r-0.5) < TOLERANCE*TOLERANCE)
+      if (std::abs(r-0.5) < distortion_tol*distortion_tol)
         return true;
 
-      return ((std::abs(R-std::round(R)) > distortion_tol) == distortion);
+      // Boundary nodes are allowed to slide along the boundary.
+      // However, nodes that are part of more than one boundary (i.e., corners) should remain fixed.
+
+      // Get boundary ids associated with the node
+      std::vector<boundary_id_type> boundary_ids;
+      boundary_info.boundary_ids(&node, boundary_ids);
```

Comment:
Here, though, you're correct to use nodal boundary ids; definitely the easiest way to write the test.